core/app/helpers/spree/currency_helper.rb

```
@@ -21,22 +26,32 @@ def should_render_currency_dropdown?
     end
 
+    # Returns the list of supported currencies for the current store.
+    # @return [Array<Money::Currency>] the list of supported currencies
     def preferred_currencies
+      Spree::Deprecation.warn('preferred_currencies is deprecated and will be removed in Spree 6.0. Use current_store.supported_currencies_list instead.')
       @preferred_currencies ||= current_store.supported_currencies_list
     end
 
     def preferred_currencies_select_options
-      preferred_currencies.map { |currency| ["#{currency.name} - #{currency.iso_code}", currency] }
+      Spree::Deprecation.warn('preferred_currencies_select_options is deprecated and will be removed in Spree 6.0. Use supported_currency_options instead.')
```

RuboCop (1.81.1)

[convention] Line is too long. [156/150]

(Layout/LineLength)